runtime/src/main/scala/tailcall/runtime/service/GraphQLGenerator.scala

```diff
@@ -8,17 +8,18 @@ import caliban.wrappers.Wrapper
 import tailcall.runtime.model.Blueprint
 import tailcall.runtime.service.DataLoader.HttpDataLoader
 import tailcall.runtime.transcoder.Transcoder
+import zio.http.model.Headers
 import zio.{ZIO, ZLayer}
 
 trait GraphQLGenerator {
-  def toGraphQL(document: Blueprint): GraphQL[HttpDataLoader]
+  def toGraphQL(document: Blueprint): GraphQL[HttpDataLoader with Headers]
 }
 
 object GraphQLGenerator {
   final case class Live(sGen: StepGenerator) extends GraphQLGenerator {
-    override def toGraphQL(blueprint: Blueprint): GraphQL[HttpDataLoader] = {
-      new GraphQL[HttpDataLoader] {
-        override protected val schemaBuilder: RootSchemaBuilder[HttpDataLoader] = {
+    override def toGraphQL(blueprint: Blueprint): GraphQL[HttpDataLoader with Headers] = {
+      new GraphQL[HttpDataLoader with Headers] {
```

Comment:
```suggestion
      new GraphQL[HttpDataLoader] {
```

Create a actual HttpDataLoader service that is not just a type alias to Dataloader[Http].
This might also be renamed to something better (later may be).